### PR TITLE
internal: Fix some test async behavior

### PR DESCRIPTION
**Pull request by ntucker**
### Motivation

- Some typos around async behavior found when doing https://github.com/coinbase/rest-hooks/pull/553

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ const baseConfig = {
     'packages/experimental',
     'packages/legacy/src/resource',
   ],
+  testEnvironment: 'jsdom',
```
Comment (ljharb) [inline jest.config.js:13]: this means you won't be running tests "like it's node", but more "like it's a browser".

i've found it works well on other projects to run tests twice, once with each testEnvironment. not a blocker ofc.

File: packages/hooks/src/__tests__/useCancelling.ts [hunk 2]
```diff
@@ -34,6 +34,10 @@ describe('useCancelling()', () => {
       .delay(2000)
       .reply(200, payload2);
   });
+  afterAll(() => {
+    jest.useRealTimers();
+    nock.cleanAll();
+  });
```
Comment (ljharb) [inline packages/hooks/src/__tests__/useCancelling.ts:40]: should this be after _all_, or after _each_?

**Review by ljharb — APPROVED** [review]

File: jest.config.js (hunk 1 above)
Comment (ntucker) [reply jest.config.js:13]: Jest config returns array. Second one is node like you said.

File: packages/hooks/src/__tests__/useCancelling.ts (hunk 2 above)
Comment (ntucker) [reply packages/hooks/src/__tests__/useCancelling.ts:40]: It was before all for the setup. Does jest interleave tests?

File: jest.config.js (hunk 1 above)
Comment (ljharb) [reply jest.config.js:13]: ah oops, didn't look at the whole file. nice :-)

File: packages/hooks/src/__tests__/useCancelling.ts (hunk 2 above)
Comment (ljharb) [reply packages/hooks/src/__tests__/useCancelling.ts:40]: given that timers and nock are per-process, using "all" means that tests in the same process (which might or might not be as granular as "same file") could be depending on the timer/nock behavior from other tests to pass. Using "each" ensures every test runs in isolation from the others in this respect.

**Pull request merged**